pkg/kubernetes-utils.go

```diff
@@ -295,6 +304,92 @@ func ExistsNamespace(useInClusterConfig bool, namespace string) (bool, error) {
 	return true, nil
 }
 
+// GetKeptnManagedNamespace returns the list of namespace with the annotation & label `keptn.sh/managed-by: keptn`
+func GetKeptnManagedNamespace(useInClusterConfig bool) ([]string, error) {
+	var namespaceList *typesv1.NamespaceList
+	var namespaces []string
+	clientset, err := GetClientset(useInClusterConfig)
+	if err != nil {
+		return nil, err
+	}
+	namespaceList, err = clientset.CoreV1().Namespaces().List(metav1.ListOptions{
+		LabelSelector: "keptn.sh/managed-by",
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, namespace := range namespaceList.Items {
+		if metav1.HasAnnotation(namespace.ObjectMeta, "keptn.sh/managed-by") {
+			namespaces = append(namespaces, namespace.GetObjectMeta().GetName())
+		}
+	}
+	return namespaces, nil
+}
+
+// PatchKeptnManagedNamespace to patch the namespace with the annotation & label `keptn.sh/managed-by: keptn`
+func PatchKeptnManagedNamespace(useInClusterConfig bool, namespace string) error {
+	var patchData = []byte(`{"metadata": {"annotations": {"keptn.sh/managed-by": "keptn"}, "labels": {"keptn.sh/managed-by": "keptn"}}}`)
+	clientset, err := GetClientset(useInClusterConfig)
+	if err != nil {
+		return err
+	}
+	_, err = clientset.CoreV1().Namespaces().Patch(namespace, types.StrategicMergePatchType, patchData)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetKeptnEndpointFromIngress returns the host of ingress object Keptn Installation
+func GetKeptnEndpointFromIngress(useInClusterConfig bool, namespace string, ingressName string) (string, error) {
+	var keptnIngress *v1beta1.Ingress
+	clientset, err := GetClientset(useInClusterConfig)
+	if err != nil {
+		return "", err
+	}
+	keptnIngress, err = clientset.ExtensionsV1beta1().Ingresses(namespace).Get(ingressName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return keptnIngress.Spec.Rules[0].Host, nil
+}
+
+// GetKeptnEndpointFromService returns the loadbalancer service IP from Keptn Installation
+func GetKeptnEndpointFromService(useInClusterConfig bool, namespace string, serviceName string) (string, error) {
+	var keptnService *typesv1.Service
+	clientset, err := GetClientset(useInClusterConfig)
+	if err != nil {
+		return "", err
+	}
+	keptnService, err = clientset.CoreV1().Services(namespace).Get(serviceName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	switch keptnService.Spec.Type {
+	case "LoadBalancer":
+		return keptnService.Status.LoadBalancer.Ingress[0].IP, nil
```

Comment:
please add a check for the length of the `keptnService.Status.LoadBalancer.Ingress` array

Reply:
Done. Please review